tpm/ak.go

```diff
@@ -0,0 +1,204 @@
+package tpm
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	"github.com/google/go-attestation/attest"
+
+	"go.step.sm/crypto/tpm/storage"
+)
+
+type AK struct {
+	Name      string
+	Data      []byte
+	CreatedAt time.Time
+
+	tpm *TPM
+}
+
+func (t *TPM) CreateAK(ctx context.Context, name string) (AK, error) {
+	result := AK{}
+	if err := t.Open(ctx); err != nil {
+		return result, fmt.Errorf("failed opening TPM: %w", err)
+	}
+	defer t.Close(ctx, false)
+
+	at, err := attest.OpenTPM(t.attestConfig)
+	if err != nil {
+		return result, fmt.Errorf("failed opening TPM: %w", err)
+	}
+	defer at.Close()
+
+	now := time.Now()
+
+	if name == "" {
```

Comment:
I think you should extract this name generation code into its own function (as it's used in a couple of places).